contracts/src/main/kotlin/com/contractsAndStates/contracts/LongestRoadContract.kt

```diff
@@ -0,0 +1,57 @@
+package com.contractsAndStates.contracts
+
+import com.contractsAndStates.states.*
+import net.corda.core.contracts.CommandData
+import net.corda.core.contracts.Contract
+import net.corda.core.contracts.requireSingleCommand
+import net.corda.core.contracts.requireThat
+import net.corda.core.transactions.LedgerTransaction
+import java.security.PublicKey
+
+class LongestRoadContract : Contract {
+
+    companion object {
+        const val ID = "com.contractsAndStates.contracts.LongestRoadContract"
+    }
+
+    override fun verify(tx: LedgerTransaction) {
+        val command = tx.commands.requireSingleCommand<Commands>()
+
+        val longestRoadInputStates = tx.inputsOfType<LongestRoadState>().first()
```

Comment:
If you expect a single `LongestRoadState`, use `.single()` instead. And, in order to have a proper error message, do:

```kotlin
"Expect a single xxx" using (tx.inputsOfType<LongestRoadState>().size == 1)
```

Reply:
Ok